### PR TITLE
Add optional serialize/deserialize functions

### PR DESCRIPTION
**Pull request by aubergene**
Sometimes you might want to store something which isn't directly serializable with `JSON.stringify`

### CODE REVIEW

**Comment by joshnuss** [comment]
Hi,

I guess we should add this. It was actually [requested before](https://github.com/joshnuss/svelte-local-storage-store/pull/75) but then I wasn't sure if many people would need it. Since there multiple requests, I think it now makes sense to add.

One thing to consider, what if instead of passing 2 functions as options (`serializer` and `deserializer`), it just took a single object that had the same shape as `JSON`?

ie 

```ts
interface Serializer {
  parse(text: string): object
  stringify(object: any): string
}

interface Options {
  serializer: Serializer
}

export function writable<T>(key: string, initialValue: T, options?: Options): Writable<T> {
  const serializer = options?.serializer || JSON
  // ...
}
```

What do you think?

**Comment by aubergene** [comment]
Yes, using the serializer interface is much nicer, thank you!

File: test/localStorageStore.test.ts [hunk 1]
```diff
@@ -178,4 +178,18 @@ describe('writable()', () => {
       unsub()
     })
   })
+
+  it('allows custom serialize/deserialize functions', () => {
+    const serializer = JSON
+
+    const s = new Set([1, 2, 3])
```
Comment (joshnuss) [inline test/localStorageStore.test.ts:185]: Also, would prefer to avoid one-letter var names.

**Comment by joshnuss** [comment]
The CI error can be fixed by adjusted the `target` in `tsconfig.json`, from `es2020` -> `es2019`

**Review by joshnuss — APPROVED** [review]

**Comment by joshnuss** [comment]
Thanks @aubergene!

**Pull request merged**